seedsource_core/django/seedsource/serializers.py

```diff
@@ -107,3 +113,32 @@ class RegionSerializer(serializers.ModelSerializer):
     class Meta:
         model = Region
         fields = ('name',)
+
+
+class ShareURLSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ShareURL
+        fields = ('configuration', 'version', 'hash', 'created', 'accessed')
+        read_only_fields = ('hash', 'created', 'accessed')
+
+    def create(self, validated_data):
+        configuration = validated_data['configuration']
+        version = validated_data['version']
+        string_to_hash = configuration + str(version)
+        hash_as_bytes = hashlib.sha256(string_to_hash.encode()).digest()
+        hash_as_integer = int.from_bytes(hash_as_bytes, sys.byteorder)
+        hash_as_b62_truncated = ''
+        for i in range(8):
+            hash_as_b62_truncated += B62_CHARS[hash_as_integer % 62]
+            hash_as_integer //= 62
+
+        attributes = {
+            'hash': hash_as_b62_truncated,
+            'configuration': configuration,
```

Comment:
The `configuration` field is JSONB type, but what gets written is a single string containing the JSON as text. Try adding `configuration` as an explicit field to this serializer, or if that doesn't work, do `json.loads(configuration)`.

Reply:
Try a JSON field first: https://www.django-rest-framework.org/api-guide/fields/#jsonfield